### PR TITLE
Add basic subpixel AA text support on Linux.

### PR DESCRIPTION
**Pull request by glennw**
This is some preliminary support for subpixel AA. Specifically:
 * Only on Linux for now.
 * No gamma correction.
 * Disabled by default (use -Z subpixel-aa to test).
 * Doesn't deal with rotations, subpixel positioning etc.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/527)

### CODE REVIEW

**Comment by jrmuizel** [comment]
Re: gamma correction. I've elaborated some of my thoughts in #230.

File: webrender/src/tiling.rs [hunk 1]
```diff
@@ -105,6 +106,25 @@ impl AlphaBatchHelpers for PrimitiveStore {
...
+                    FontRenderMode::Subpixel => BlendMode::Subpixel(text_run_cpu.color),
+                    FontRenderMode::Alpha | FontRenderMode::Mono => BlendMode::Alpha,
+                }
+            }
+            _ => {
+                if needs_blending {
+                    BlendMode::Alpha
+                } else {
+                    BlendMode::None
+                }
```
Comment (pcwalton) [inline webrender/src/tiling.rs]: nit: could be

    _ if needs_blending => BlendMode::Alpha,
    _ => BlendMode::None

Up to you

File: webrender/src/platform/unix/font.rs [hunk 2]
```diff
@@ -215,6 +217,9 @@ impl FontContext {
                             }
                         }
                         FT_PIXEL_MODE_LCD => {
+                            // Extra subpixel on each side of the glyph.
+                            glyph_width += 2;
+
```
Comment (pcwalton) [inline webrender/src/platform/unix/font.rs]: Ugly. :( Oh well, probably can't be helped.

File: webrender/src/renderer.rs [hunk 3]
```diff
@@ -92,6 +92,7 @@ const GPU_TAG_PRIM_BORDER: GpuProfileTag = GpuProfileTag { label: "Border", colo
 pub enum BlendMode {
     None,
     Alpha,
+    Subpixel(ColorF),
```
Comment (pcwalton) [inline webrender/src/renderer.rs]: Could you add a comment saying that the `ColorF` is the color of the text?

File: webrender/src/tiling.rs (hunk 1 above)
Comment (glennw) [inline webrender/src/tiling.rs]: I've left as is for now - will tidy up when I expand this function today for a different PR.

File: webrender/src/renderer.rs (hunk 3 above)
Comment (glennw) [inline webrender/src/renderer.rs]: Fixed

**Pull request merged**